workspaces/ui-v2/src/components/ShapeRenderer/ShapeRowBase.tsx

```diff
@@ -119,22 +119,22 @@ export const RenderRootShape = ({
   shape,
   right,
 }: {
-  shape: IShapeRenderer[];
+  shape: IShapeRenderer;
   right?: any[];
 }) => {
   const { depth } = useDepth();
   return (
     <>
-      <ShapeRowBase depth={depth}>
-        <RenderFieldLeadingValue shapeRenderers={shape} />
+      <ShapeRowBase depth={depth} changes={shape.changes}>
+        <RenderFieldLeadingValue shapeRenderers={[shape]} />
         {right ? (
           <>
             <div style={{ flex: 1 }} />
             {right}
           </>
         ) : null}
       </ShapeRowBase>
-      <RenderFieldRowValues shapeRenderers={shape} />
+      <RenderFieldRowValues shapeRenderers={[shape]} />
```

Comment:
There's more we can do in these components since these components are designed to handle only 1 shape. I'm not sure if it's possible for shape to be optional (i.e. IShapeRenderer | null)